Source/DafnyCore/Options/TranslationRecord.cs

```diff
@@ -26,18 +26,11 @@ public class TranslationRecord {
 
     OptionsByModule = new();
 
-    foreach (var module in program.CompileModules) {
+    foreach (var module in program.RawModules()) {
       if (module is DefaultModuleDefinition || !module.ShouldCompile(program.Compilation)) {
         continue;
       }
 
-      // This is primarily here to exclude prefix modules
-      // (e.g. something like A.B that only appears in a module A.B.C { ... } declaration)
-      // since those can appear in multiple separately-compiled projects. 
-      if (ModuleEmptyForCompilation(module)) {
-        continue;
-      }
-
```

Comment:
Mistake, I'll add this back.

I reverted this commit: https://github.com/dafny-lang/dafny/commit/4ce2f7e99e8c1ba6cdaa104e9943036587f703ce

That commit prevented writing files if they were empty.
However, Python code generation still generates imports for those files.
These imports aren't valid since the files don't exist.
I would bet that other languages have the same issue, but haven't tested this.
We'd need a fix for this, since the current state doesn't work.